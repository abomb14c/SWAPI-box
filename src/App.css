.App {
  text-align: center;
}

.App-header {
  font-size: 36px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.data-button {
  height: 40px;
  width: 100px;
  margin-bottom: 15px;
  background: none;
  color: #fff;
  font-size: 16px;
  outline: none;
}

.data-button:hover {
  background-color: #EB544D;
}

.data-button:active {
  background-color: #EB544D;
  outline: none;
}