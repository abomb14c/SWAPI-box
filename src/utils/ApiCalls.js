 
  const getPeopleData = async () => {
    const url = 'https://swapi.co/api/people/'
    const response = await fetch(url)
    const data = await response.json()
    const people = await data.results.map (async person => {
      const homePlanet = await getHomeWorld(person.homeworld)
      const species= await getSpecies(person.species)
      return {
        name: person.name,
        ...homePlanet,
        species
      }
    }) 
    return Promise.all(people)
  }

  const getHomeWorld = async (url) => {
      const response = await fetch(url)
      const planet = await response.json()
      return {planet:planet.name, population:planet.population}
  }
  
  const getSpecies = async (url) => {
      const response = await fetch(url)
      const species = await response.json()
      return species.name
  }

  const getPlanets = async () => {
    const url = 'https://swapi.co/api/planets'
    const response = await fetch(url)
    const data = await response.json()
    const planets = await data.results.map(async planet => {
      const planetResidents = await getPlanetPeople(planet.residents)
      return{
        name: planet.name,
        terrain: planet.terrain,
        population: planet.population,
        climate: planet.climate,
        residents: planetResidents
      }
    })
    
    return Promise.all(planets)
  }

  const getPlanetPeople = (url) => {
    const unresolvedPromises = url.map(async result => {
        const response = await fetch(result)
        const residents= await response.json()

        return residents.name
      })
    return Promise.all(unresolvedPromises)
  }

  const getVehicles = async () => {
    const url = 'https://swapi.co/api/vehicles/'
    const response = await fetch(url)
    const data = await response.json()
    const vehicles = await data.results.map(async vehicle => {
      return {
        name: vehicle.name,
        model: vehicle.model,
        type: vehicle.vehicle_class,
        passengers: vehicle.passengers
      }
    })
    return Promise.all(vehicles)
  }

  export {
      getPeopleData,
      getPlanets,
      getVehicles
  }
