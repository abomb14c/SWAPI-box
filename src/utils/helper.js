
 const getCrawlText= (data) => {
  const results = data.results.map(result => {
    return {
       crawlText: result.opening_crawl, 
       title: result.title,
       release: result.release_date
    }
  })
  return getRandom(results)
}

const getRandom= (results) => {
    return results[Math.floor(Math.random()*results.length)]
}


export  {getCrawlText

}