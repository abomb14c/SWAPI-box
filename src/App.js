import React, { Component } from 'react';
import './App.css';
import {getCrawlText} from './utils/helper.js'
import { CrawlText } from './CrawlText/CrawlText'
import {getPeopleData, getPlanets, getVehicles} from './utils/ApiCalls'
import CardSection from './CardSection/CardSection'

class App extends Component {
  constructor(props) {
    super(props)

    this.state = {
      crawlData: [],
      cleanPeople: [],
      cleanPlanets: [],
      cleanVehicles: [],
      activeCategory: "",
      favoritesArray: []
    }
  }
  
  setPeopleData = async () => {
    const cleanPeople = await getPeopleData()
    this.setState({cleanPeople,
      crawlData: [],
      activeCategory: "is people",
      cleanPlanets: [],
      cleanVehicles: []
    })
  }

  findFavoritePerson = (id) => {
    const favoritePerson = this.state.cleanPeople.find(person => person.id === id)
    this.setState({
      favoritesArray:[...this.state.favoritesArray, favoritePerson]
    })
  }

  findFavoritePlanet = (id) => {
    const favoritePlanet = this.state.cleanPlanets.find(planet => planet.id === id)
    this.setState({
      favoritesArray:[...this.state.favoritesArray, favoritePlanet]
    })
  }

  findFavoriteVehicle = (id) => {
    const favoriteVehicle = this.state.cleanVehicles.find(vehicle => vehicle.id === id)
    this.setState({
      favoritesArray:[...this.state.favoritesArray, favoriteVehicle]
    })
  }

  setPlanetData = async () => {
    const cleanPlanets = await getPlanets()
    this.setState({cleanPlanets,
      crawlData:[],
      activeCategory:"is planets",
      cleanPeople: [],
      cleanVehicles: []
    })
  }

  setVehicleData = async () => {
    const cleanVehicles = await getVehicles()
    this.setState({cleanVehicles,
      crawlData:[],
      activeCategory: "is vehicles",
      cleanPeople: [],
      cleanPlanets: []
    })
  }

  setFavorites = () => {
    this.setState({
      activeCategory: "is favorites"
    })
  }

  componentDidMount() {
    const url = 'https://swapi.co/api/films/'
    fetch(url)
    .then(response => response.json())
    .then(data => {
     const crawlData= getCrawlText(data)
      this.setState({crawlData})
    })
  }
  
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">SWAPI-Box</h1>
        </header>
        <div className="button-container">
            <button className="data-button" onClick={this.setPeopleData}>people</button>
            <button className="data-button" onClick={this.setPlanetData}>planets</button>
            <button className="data-button" onClick={this.setVehicleData}>vehicles</button>
            <button className="data-button" onClick={this.setFavorites}>Favorites</button>
        </div>
        {this.state.crawlData &&
          <CrawlText data={this.state.crawlData} />
        }

        {this.state.activeCategory.includes("is")  &&

        <CardSection activeCategory={this.state.activeCategory} 
        people={this.state.cleanPeople}
        planets={this.state.cleanPlanets}
        vehicles={this.state.cleanVehicles}
        findFavoritePerson={this.findFavoritePerson}
        findFavoritePlanet={this.findFavoritePlanet}
        findFavoriteVehicle={this.findFavoriteVehicle}
        favoritesArray={this.state.favoritesArray}

        />
      }
      </div>
    );
  }
}

export default App;
