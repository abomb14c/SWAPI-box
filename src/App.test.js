import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {shallow, mount} from 'enzyme';
import * as ApiCalls from './utils/ApiCalls';
import * as mockData from "./utils/mockData";


describe('App', () => {
  let wrapper;
  let getPeopleData;


  beforeEach(() => {
    wrapper = shallow(<App />, {disableLifecycleMethods: true});
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('should have an initial state', () => {
    expect(wrapper.state().crawlData).toEqual([]);
    expect(wrapper.state().cleanPeople).toEqual([]);
    expect(wrapper.state().cleanPlanets).toEqual([]);
    expect(wrapper.state().cleanVehicles).toEqual([]);
    expect(wrapper.state().activeCategory).toBe("");
    expect(wrapper.state().favoritesArray).toEqual([]);
  })
describe('setPeopleData', () => {
  
  it('should call getPeople data', async () => {
    ApiCalls.getPeopleData= jest.fn().mockReturnValue(mockData.fetchedPeopleData);
    
    await wrapper.instance().setPeopleData()
    
    expect(ApiCalls.getPeopleData).toHaveBeenCalled()
  })
  
  it('should set state for people', async() => {
    const expected = {
      activeCategory: "is people",
      id:`people0`,
      name: "Luke",
      planet: "Tattoine",
      population: 20000,
      species: "human"
    }
    
    ApiCalls.getPeopleData= jest.fn().mockReturnValue(expected);
    await wrapper.instance().setPeopleData()
    
    expect(wrapper.instance().state.cleanPeople).toEqual(expected)
  })
})
;
describe('setPlanetData', () => {
  it('should call getPlanets', async () => {
    ApiCalls.getPlanets= jest.fn().mockReturnValue(mockData.fetchedPlanetData);
    
    await wrapper.instance().setPlanetData()
    
    expect(ApiCalls.getPlanets).toHaveBeenCalled()
  })
  it('should set state for planets', async() => {
    const expected = {
      activeCategory: "is planets",
      id:`planet0`,
      name: "Tattoine",
      terrain: "grassy, bro",
      climate: "warm af",
      residents: 40000
    }
    
    ApiCalls.getPlanets= jest.fn().mockReturnValue(expected);
    await wrapper.instance().setPlanetData()
    expect(wrapper.instance().state.cleanPlanets).toEqual(expected)
  })
});

  describe('setVehicleData', () => {
    it('should call getVehicles', async () => {
      ApiCalls.getVehicles= jest.fn().mockReturnValue(mockData.fetchedVehicles);
      
      await wrapper.instance().setVehicleData()
      
      expect(ApiCalls.getVehicles).toHaveBeenCalled()
    })
    it('should set state for vehicles', async() => {
      const expected = {
        activeCategory: "is vehicles",
        id:`vehicle0`,
        name: "Crawler",
        model: "S Class",
        type: "Pickup",
        passengers: 40000
      }
      
      ApiCalls.getVehicles= jest.fn().mockReturnValue(expected);
      await wrapper.instance().setVehicleData()
      expect(wrapper.instance().state.cleanVehicles).toEqual(expected)
    })
    
  });

  describe('find methods', () => {
    
    it('should find favorite person', () => {
      wrapper.setState({ cleanPeople: [{ id: 'Person0' }] })
      const expected = [{ id: 'Person0' }];

      wrapper.instance().findFavoritePerson('Person0');

      expect(wrapper.state().favoritesArray).toEqual(expected);
    });

    it('should find favorite planet', () => {
      wrapper.setState({ cleanPlanets: [{ id: 'Planet0' }] })
      const expected = [{ id: 'Planet0' }];

      wrapper.instance().findFavoritePlanet('Planet0');

      expect(wrapper.state().favoritesArray).toEqual(expected);
    });
    it('should find favorite vehicles', () => {
      wrapper.setState({ cleanVehicles: [{ id: 'Vehicle0' }] })
      const expected = [{ id: 'Vehicle0' }];

      wrapper.instance().findFavoriteVehicle('Vehicle0');

      expect(wrapper.state().favoritesArray).toEqual(expected);
    });
   
    });
  });



