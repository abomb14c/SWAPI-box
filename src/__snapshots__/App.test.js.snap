// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    SWAPI-Box
          </h1>
</header>,
        <div
          className="button-container"
>
          <button
                    className="data-button"
                    onClick={[Function]}
          >
                    people
          </button>
          <button
                    className="data-button"
                    onClick={[Function]}
          >
                    planets
          </button>
          <button
                    className="data-button"
                    onClick={[Function]}
          >
                    vehicles
          </button>
          <button
                    className="data-button"
                    onClick={[Function]}
          >
                    Favorites
          </button>
</div>,
        <Unknown
          data={Array []}
/>,
        false,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="App-title"
>
            SWAPI-Box
</h1>,
          "className": "App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWAPI-Box",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "SWAPI-Box",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="data-button"
              onClick={[Function]}
>
              people
</button>,
            <button
              className="data-button"
              onClick={[Function]}
>
              planets
</button>,
            <button
              className="data-button"
              onClick={[Function]}
>
              vehicles
</button>,
            <button
              className="data-button"
              onClick={[Function]}
>
              Favorites
</button>,
          ],
          "className": "button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "people",
              "className": "data-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "people",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "planets",
              "className": "data-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "planets",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "vehicles",
              "className": "data-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "vehicles",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorites",
              "className": "data-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Favorites",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        SWAPI-Box
            </h1>
</header>,
          <div
            className="button-container"
>
            <button
                        className="data-button"
                        onClick={[Function]}
            >
                        people
            </button>
            <button
                        className="data-button"
                        onClick={[Function]}
            >
                        planets
            </button>
            <button
                        className="data-button"
                        onClick={[Function]}
            >
                        vehicles
            </button>
            <button
                        className="data-button"
                        onClick={[Function]}
            >
                        Favorites
            </button>
</div>,
          <Unknown
            data={Array []}
/>,
          false,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="App-title"
>
              SWAPI-Box
</h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SWAPI-Box",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "SWAPI-Box",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="data-button"
                onClick={[Function]}
>
                people
</button>,
              <button
                className="data-button"
                onClick={[Function]}
>
                planets
</button>,
              <button
                className="data-button"
                onClick={[Function]}
>
                vehicles
</button>,
              <button
                className="data-button"
                onClick={[Function]}
>
                Favorites
</button>,
            ],
            "className": "button-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "people",
                "className": "data-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "people",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "planets",
                "className": "data-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "planets",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "vehicles",
                "className": "data-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "vehicles",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorites",
                "className": "data-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Favorites",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
